apps/meteor/ee/server/models/raw/LivechatRooms.ts

```diff
@@ -293,4 +318,417 @@ export class LivechatRoomsRawEE extends LivechatRoomsRaw implements ILivechatRoo
 		queriesLogger.debug({ msg: 'LivechatRoomsRawEE.updateMany', query: restrictedQuery });
 		return super.updateMany(restrictedQuery, ...restArgs);
 	}
+
+	getConversationsBySource(start: Date, end: Date, extraQuery: Filter<IOmnichannelRoom>): AggregationCursor<ReportResult> {
+		return this.col.aggregate(
+			[
+				{
+					$match: {
+						source: {
+							$exists: true,
+						},
+						t: 'l',
+						ts: {
+							$gte: start,
+							$lt: end,
+						},
+						...extraQuery,
+					},
+				},
+				{
+					$group: {
+						_id: '$source',
+						value: { $sum: 1 },
+					},
+				},
+				{
+					$sort: { value: -1 },
+				},
+				{
+					$group: {
+						_id: null,
+						total: { $sum: '$value' },
+						data: {
+							$push: {
+								label: {
+									$ifNull: ['$_id.alias', '$_id.type'],
+								},
+								value: '$value',
+							},
+						},
+					},
+				},
+				{
+					$project: {
+						_id: 0,
+					},
+				},
+			],
+			{ hint: 'source_1_ts_1', readPreference: readSecondaryPreferred() },
```

Comment:
Hey, @sampaiodiego @MarcosSpessatto , I'm hinting the indexes here so mongo uses them.

Mongo basically sees a `$exists: true` and gets the first sparse index that can fulfill the query, even if the partialFilterExpression matches perfectly, which is a bit troll.

The extra filters for monitors should still apply under the index wings (as they filter by departmentId too) so we should be good on that sense.

Adding this comment so you know, and you can provide any ideas you may have around this and how to get better indexes :awesome: